run_batch_general.py

```diff
@@ -1,213 +1,516 @@
+                                "pipeline": posixpath.join(batchpath, batchfile),
+                                "output": outpath,
+                                "output_structure": outputstructure,
+                                "input": inputpath,
+                                "data_file": posixpath.join(batchpath, batchfile),
+                            }
+                            analysisqueue.scheduleBatch(templateMessage_analysis)
+
+        print("Analysis job submitted. Check your queue")
+
+    else:
+        print(f"Step {step} not supported.")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Make jobs for Distributed-CellProfiler"
+    )
+    parser.add_argument(
+        "step",
+        help="Step to make jobs for. Supported steps are zproj, illum, qc, qc_persite, assaydev, analysis",
+    )
+    parser.add_argument(
+        "path_style",
+        help="Style of input/output path. default or cpg (for Cell Painting Gallery structure).",
+    )
+    parser.add_argument("identifier", help="Project identifier")
+    parser.add_argument("batch", help="Name of batch")
+    parser.add_argument("platelist", type=lambda s: list(s.split(",")), help="List of plates to process")
+    parser.add_argument(
+        "--source",
+        dest="source",
+        default="source_4",
+        help="For Cell Painting Gallery, what is the source (nesting under project identifier).",
+    )
+    parser.add_argument(
+        "--plate-format",
+        dest="plate_format",
+        default=384,
+        help="Plate format. Suppports 384 or 96. Auto-generates rows and columns and will overwrite --rows and --columns.",
+    )
+    parser.add_argument(
+        "--rows",
+        dest="rows",
+        type=lambda s: list(s.split(",")),
+        default="A,B,C,D,E,F,G,H,I,J,K,L,M,N,O,P",
+        help="List of rows to process",
+    )
+    parser.add_argument(
+        "--columns",
+        dest="columns",
+        type=lambda s: list(s.split(",")),
+        default="1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24",
+        help="List of rows to process",
+    )
+    parser.add_argument(
+        "--sites",
+        dest="sites",
+        type=lambda s: list(s.split(",")),
+        default="1,2,3,4,5,6,7,8,9",
+        help="List of site to process",
+    )
+    parser.add_argument(
+        "--no-well-digit-pad",
+        dest="well_digit_pad",
+        action="store_false",
+        default=True,
+        help="Format wells with padding e.g. A01",
+    )
+    parser.add_argument(
+        "--pipeline",
+        dest="pipeline",
+        default="",
+        help="Name of the pipeline to overwrite defaults of Zproj.cppipe, illum.cppipe, qc.cppipe, assaydev.cppipe, analysis.cppipe.",
+    )
+    parser.add_argument(
+        "--outputstructure",
```

Comment:
Minor preference for this to be output-structure, but I don't feel strongly about it if you'd rather not